src/orders/service/OrdersCrudService.ts

```diff
@@ -0,0 +1,193 @@
+import Order from '../model/Order';
+import OrderCreateRequest from '../model/OrderCreateRequest';
+import OrderStatus from '../model/OrderStatus';
+import OrdersPage from '../model/OrdersPage';
+import OrdersQueryRequest from '../model/OrdersQueryRequest';
+import OrdersQueryRequestByBusinessId from '../model/OrdersQueryRequestByBusinessId';
+import OrdersQueryRequestByChannel from '../model/OrdersQueryRequestByChannel';
+import OrdersQueryRequestByChannelOrderId from '../model/OrdersQueryRequestByChannelOrderId';
+import LineItem from '../model/LineItem';
+import Fulfillment from '../model/Fulfillment';
+import RequestClientWrapper from '../../RequestClientWrapper';
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+import Resource from '../../model/Resource';
+import Version from '../../model/Version';
+import RequestCallbackParams from '../../model/RequestCallbackParams';
+import GenerateApiError from '../../utils/GenerateApiError';
+import * as Q from 'q';
+import GenericOrdersQueryRequest from './model/GenericOrdersQueryRequest';
+
+const EXPECTED_GET_STATUS = 200;
+const EXPECTED_CREATE_STATUS = 202;
+const EXPECTED_UPDATE_STATUS = 202;
+
+export default class OrdersCrudService {
+
+  constructor(private readonly client: RequestClientWrapper) { }
+
+  public get(orderId: string): Promise<Order>;
+  public get(ordersRequestByBusinessId: OrdersQueryRequestByBusinessId): Promise<Order[]>;
+  public get(ordersRequestByChannel: OrdersQueryRequestByChannel): Promise<Order[]>;
+  public get(ordersRequestByChannelOrderId: OrdersQueryRequestByChannelOrderId): Promise<Order[]>;
+  public get(orderIdOrRequest: string | GenericOrdersQueryRequest): Promise<Order> | Promise<Order[]> {
+    if (typeof orderIdOrRequest === 'string') {
+      return this.getByOrderId(orderIdOrRequest);
+    }
+    const deferred: Q.Deferred<Order[]> = Q.defer<Order[]>();
+    const getSinglePage = false;
+    this.getOrdersByRequest(orderIdOrRequest, [], deferred, getSinglePage);
+    return deferred.promise as any;
+  }
+
+  public getPage(ordersRequestByBusinessId: OrdersQueryRequestByBusinessId): Promise<OrdersPage>;
+  public getPage(ordersRequestByChannel: OrdersQueryRequestByChannel): Promise<OrdersPage>;
+  public getPage(ordersRequestByChannelOrderId: OrdersQueryRequestByChannelOrderId): Promise<OrdersPage>;
+  public getPage(orderRequest: OrdersQueryRequestByBusinessId | OrdersQueryRequestByChannel |
+    OrdersQueryRequestByChannelOrderId): Promise<OrdersPage> {
+    const deferred = Q.defer<OrdersPage>();
+    const getSinglePage = true;
+    this.getOrdersByRequest(orderRequest, [], deferred, getSinglePage);
+    return deferred.promise as any;
+  }
+
+  public update(order: Order): Promise<Order> {
```

Comment:
Somewhere within this function `X-Channel-Ape-Action-Id` should be passed as a header when the optional actionId field is populated.